### PR TITLE
new cask: CellDesigner

### PR DESCRIPTION
**Pull request by achillesrasquinha**
After making all changes to the cask:

- [x] `brew cask audit --download {{cask_file}}` is error-free.
- [x] `brew cask style --fix {{cask_file}}` reports no offenses.
- [x] The commit message includes the cask’s name and version.
- [x] The submission is for [a stable version](https://github.com/Homebrew/homebrew-cask/blob/master/doc/development/adding_a_cask.md#finding-a-home-for-your-cask) or [documented exception](https://github.com/Homebrew/homebrew-cask/blob/master/doc/development/adding_a_cask.md#but-there-is-no-stable-version).

Additionally, if **adding a new cask**:

- [x] Named the cask according to the [token reference].
- [x] `brew cask install {{cask_file}}` worked successfully.
- [x] `brew cask uninstall {{cask_file}}` worked successfully.
- [x] Checked there are no [open pull requests] for the same cask.
- [x] Checked the cask was not [already refused].
- [x] Checked the cask is submitted to [the correct repo].

[token reference]: https://github.com/Homebrew/homebrew-cask/blob/master/doc/cask_language_reference/token_reference.md
[open pull requests]: https://github.com/Homebrew/homebrew-cask/pulls
[already refused]: https://github.com/Homebrew/homebrew-cask/search?q=is%3Aclosed&type=Issues
[the correct repo]: https://github.com/Homebrew/homebrew-cask/blob/master/doc/development/adding_a_cask.md#finding-a-home-for-your-cask
[version-checksum]: https://github.com/Homebrew/homebrew-cask/blob/master/doc/cask_language_reference/stanzas/sha256.md#updating-the-sha256

### CODE REVIEW

File: Casks/celldesigner.rb [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+  sha256 'b23708f82842f4c138c2ba8d2ecd7d94942c9cedf05bcf6cc6b683e11305209c'
+
+  # systems-biology.org was verified as official when first introduced to the cask
+  url "http://www.systems-biology.org/download/CellDesigner-#{version}-osx-installer.dmg"
+  appcast 'http://systems-biology.org/software/celldesigner/'
+  name 'CellDesigner'
+  homepage 'http://www.celldesigner.org/'
+
+  depends_on cask: 'java'
+  container type: :dmg
```
Comment (vitorgalvao) [inline Casks/celldesigner.rb]: Have you tried without this line? It should not be needed.

File: Casks/celldesigner.rb [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+  version '4.4.2'
+  sha256 'b23708f82842f4c138c2ba8d2ecd7d94942c9cedf05bcf6cc6b683e11305209c'
+
+  # systems-biology.org was verified as official when first introduced to the cask
+  url "http://www.systems-biology.org/download/CellDesigner-#{version}-osx-installer.dmg"
+  appcast 'http://systems-biology.org/software/celldesigner/'
+  name 'CellDesigner'
+  homepage 'http://www.celldesigner.org/'
+
+  depends_on cask: 'java'
```
Comment (vitorgalvao) [inline Casks/celldesigner.rb]: Should be a [`caveat`](https://github.com/Homebrew/homebrew-cask/blob/master/doc/cask_language_reference/stanzas/caveats.md#caveats-mini-dsl) instead.

**Review by vitorgalvao — CHANGES_REQUESTED** [review]

File: Casks/celldesigner.rb (hunk 1 above)
Comment (achillesrasquinha) [reply Casks/celldesigner.rb]: Yes. Alright.

**Comment by achillesrasquinha** [comment]
@vitorgalvao fixed all changes.

File: Casks/celldesigner.rb [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+  zap trash: [
+               "/Applications/CellDesigner#{version}",
+               '~/.celldesigner',
+               '~/.copasi',
+               '~/.sbw',
+               '~/.oracle_jre_usage',
+             ]
+
+  caveats do
+    depends_on_java
```
Comment (vitorgalvao) [inline Casks/celldesigner.rb]: It needs a version as the argument. What versions does this require?

**Review by vitorgalvao — CHANGES_REQUESTED** [review]

File: Casks/celldesigner.rb (hunk 3 above)
Comment (achillesrasquinha) [reply Casks/celldesigner.rb]: '6+'

**Comment by vitorgalvao** [comment]
Thank you, @achillesrasquinha.

By the way, [that’s a great last name](https://glosbe.com/pt/en/%C3%A0%20rasquinha).

**Pull request merged**